test/study-webidl.js

```diff
@@ -361,12 +391,318 @@ interface mixin MyNamespaceMixin {};
+};
+
+interface mixin MyPartialRoom {
+  Promise<DOMString> overload();
+};
+
+partial interface mixin MyPartialRoom {
+  Promise<DOMString> overload(DOMString thing);
+};
+`);
+    assertEqual(report.length, 2);
+    assertEqual(report, 0, 'name', 'overloaded');
+    assertEqual(report, 0, 'message', '"operation overload" in interface "MyHome" overloads an operation defined in interface mixin "MyRoom"');
+    assertEqual(report, 1, 'name', 'overloaded');
+    assertEqual(report, 1, 'message', '"operation overload" in partial interface mixin "MyPartialRoom" overloads an operation defined in interface mixin "MyPartialRoom"');
+  });
+
+
+  it('reports overloads across definitions (mixin, different specs)', async () => {
+     const crawlResult = toCrawlResult(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
+};
+MyHome includes MyRoom;
+
+interface mixin MyPartialRoom {
+  Promise<DOMString> overload();
+};
+
+partial interface mixin MyPartialRoom {
+  Promise<DOMString> overload(DOMString thing);
+};
+`).concat(toCrawlResult(`
+interface mixin MyRoom {
+  undefined overload(DOMString thing);
+};
+
+partial interface mixin MyPartialRoom {
+  Promise<DOMString> overload(DOMString thing, boolean asap);
+};
+`, specUrl2));
+    const report = await studyWebIdl(crawlResult);
+
+    assertEqual(report.length, 4);
+    assertEqual(report, 0, 'name', 'overloaded');
+    assertEqual(report, 0, 'message', `"operation overload" in interface "MyHome" overloads an operation defined in interface mixin "MyRoom" (in ${specUrl2})`);
+    assertEqual(report, 0, 'specs.length', 1);
+    assertEqual(report, 0, 'specs[0].url', specUrl);
+
+    assertEqual(report, 1, 'name', 'overloaded');
+    assertEqual(report, 1, 'message', '"operation overload" in partial interface mixin "MyPartialRoom" overloads an operation defined in interface mixin "MyPartialRoom"');
+    assertEqual(report, 1, 'specs.length', 1);
+    assertEqual(report, 1, 'specs[0].url', specUrl);
+
+    assertEqual(report, 2, 'name', 'overloaded');
+    assertEqual(report, 2, 'message', `"operation overload" in partial interface mixin "MyPartialRoom" overloads an operation defined in interface mixin "MyPartialRoom" (in ${specUrl})`);
+    assertEqual(report, 2, 'specs.length', 1);
+    assertEqual(report, 2, 'specs[0].url', specUrl2);
+
+    assertEqual(report, 3, 'name', 'overloaded');
+    assertEqual(report, 3, 'message', `"operation overload" in partial interface mixin "MyPartialRoom" (in ${specUrl2}) overloads an operation defined in another partial interface mixin "MyPartialRoom" (in ${specUrl})`);
+    assertEqual(report, 3, 'specs.length', 2);
+    assertEqual(report, 3, 'specs[0].url', specUrl2);
+    assertEqual(report, 3, 'specs[1].url', specUrl);
+  });
+
+
+  it('reports overloads across definitions (partial and mixin, same spec)', async () => {
+    const report = await analyzeIdl(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
```

Comment:
the test could be made a bit shorter while still achieving its goal (IIUC) by not including `overload()`  in the main interface (and the main mixin)